src/translate.py

```diff
@@ -996,6 +996,16 @@ def format(self, fmt: Formatter) -> str:
                 type=self.type,
             ).format(fmt)
 
+        # For comparisons to a Literal, cast the Literal to the type of the lhs
+        # (This is not done with complex expressions to avoid propagating incorrect
+        # type information: end-of-array pointers are particularly bad.)
+        if self.is_comparison() and isinstance(right_expr, Literal):
```

Comment:
We probably don't want enum names or NULL to show up for </>/<=/>= comparisons. Can we move the `self.op in ("==", "!=")` check to up here?

Reply:
ah, I thought you were concerned with </>/<=/>= & U; this sounds good then.